ledger-tool/src/main.rs

```diff
@@ -2965,3 +2996,110 @@ fn main() {
     measure_total_execution_time.stop();
     info!("{}", measure_total_execution_time);
 }
+
+fn record_transactions(
+    recv: crossbeam_channel::Receiver<TransactionStatusMessage>,
+    filename: &Path,
+) {
+    #[derive(Serialize, Deserialize, Default, Clone)]
+    struct Transaction {
+        index: usize,
```

Comment:
I guess having the index solves a previous problem of parallel execution potentially changing the order in which transactions land. And then it looks like you repeatedly insert / sort the structure by index below ?

Reply:
Yes, that's right.

It's probably cleaner to do the sort just once, so I've moved that.